core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -154,25 +160,70 @@ public void onTimelineChanged(Timeline timeline, @Nullable Object manifest, int
         if (player != null) {
             adGroupIndex = player.getCurrentAdGroupIndex();
             adIndexInGroup = player.getCurrentAdIndexInAdGroup();
+
+            if (reason == Player.TIMELINE_CHANGE_REASON_PREPARED && adGroupIndex != -1 && adIndexInGroup != -1) {
```

Comment:
not a blocker: but what about extracting a method for this check?
```
adGroupIndex != -1 && adIndexInGroup != -1
```
it's used in few places and I wonder if we could make it more understandable while reducing some duplication